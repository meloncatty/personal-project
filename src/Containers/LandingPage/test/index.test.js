import React from 'react'
import { shallow } from 'enzyme'
import { LandingPage, mapDispatchToProps } from '../'

describe('LandingPage', () => {
  let mockProps
  let landingPage

  beforeEach(() => {
    mockProps = {
      fetchArticles: jest.fn()
    }
    landingPage = shallow(<LandingPage {...mockProps}/>) 
  })

  it('should match snapshot', () => {
    expect(landingPage).toMatchSnapshot()
  })

  it('should have default state', () => {
    const expected = {
      searchInput: '',
      redirectToSearch: false
    }

    expect(landingPage.state()).toEqual(expected)
  })

  it('should handle change upon input change', () => {
    const mockEvent = { target: { value: 'query' } }

    landingPage.find('input').simulate('change', mockEvent)

    expect(landingPage.state('searchInput')).toEqual('query')
  })

  it('should call fetchArticles when form is submitted', () => {
    const mockFetchArticles = (landingPage.instance().handleSubmit = jest.fn())
    const mockPreventDefault = {preventDefault: jest.fn()}

    landingPage.find('button').simulate('submit', mockPreventDefault)
    landingPage.instance().handleSubmit()

    expect(mockFetchArticles).toHaveBeenCalledTimes(1)
  })

  it('should update redirectToSearch when handleSubmit is called', () => {
    const mockPreventDefault = {preventDefault: jest.fn()}

    landingPage.find('form').simulate('submit', mockPreventDefault)

    expect(landingPage.state('redirectToSearch')).toBe(true)
  })

  describe('handleSubmit', () => {
    it('should call fetchArticles with correct parameters', () => {
      const mockPreventDefault = {preventDefault: jest.fn()}

      landingPage.setState({ searchInput: 'aliens' })
      landingPage.find('form').simulate('submit', mockPreventDefault)

      expect(landingPage.instance().props.fetchArticles).toHaveBeenCalledWith('aliens')
    })

    it('should update state of component', () => {
      const expected = {
        redirectToSearch: true,
        searchInput: ''
      }      
      const mockPreventDefault = {preventDefault: jest.fn()}
      landingPage.setState({ searchInput: 'aliens' })
      landingPage.find('form').simulate('submit', mockPreventDefault)

      expect(landingPage.state()).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('should return an object', () => {
      const mockDispatch = jest.fn()
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.fetchArticles('aliens')

      expect(mockDispatch).toHaveBeenCalled()
    })
  })
})
