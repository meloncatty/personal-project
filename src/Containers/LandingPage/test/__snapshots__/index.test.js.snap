// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LandingPage
    fetchArticles={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          onSubmit={[Function]}
>
          <input
                    className="search-input"
                    name="query"
                    onChange={[Function]}
                    placeholder="Search through thousands of open source articles"
                    type="text"
                    value=""
          />
          <button
                    className="submit-search"
          >
                    Submit
          </button>
</form>,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="search-input"
              name="query"
              onChange={[Function]}
              placeholder="Search through thousands of open source articles"
              type="text"
              value=""
/>,
            <button
              className="submit-search"
>
              Submit
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "search-input",
              "name": "query",
              "onChange": [Function],
              "placeholder": "Search through thousands of open source articles",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
              "className": "submit-search",
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
        ],
        "type": "form",
      },
      false,
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            onSubmit={[Function]}
>
            <input
                        className="search-input"
                        name="query"
                        onChange={[Function]}
                        placeholder="Search through thousands of open source articles"
                        type="text"
                        value=""
            />
            <button
                        className="submit-search"
            >
                        Submit
            </button>
</form>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="search-input"
                name="query"
                onChange={[Function]}
                placeholder="Search through thousands of open source articles"
                type="text"
                value=""
/>,
              <button
                className="submit-search"
>
                Submit
</button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "search-input",
                "name": "query",
                "onChange": [Function],
                "placeholder": "Search through thousands of open source articles",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
                "className": "submit-search",
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
          ],
          "type": "form",
        },
        false,
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
